alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionEvaluator.java

```diff
@@ -0,0 +1,312 @@
+                case avg:
+                    return data.getAvg().doubleValue();
+                case max:
+                    return data.getMax().doubleValue();
+                case median:
+                    return data.getMedian().doubleValue();
+                case min:
+                    return data.getMin().doubleValue();
+                case samples:
+                    return data.getSamples() * 1.0;
+                case sum:
+                    return data.getSum().doubleValue();
+                default:
+                    throw new IllegalArgumentException("Unexpected func [" + func + "]");
+            }
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid value for func [" + func + "] in " + data.toString());
+        }
+    }
+
+    private double getAvailabilityValue(String func, AvailabilityBucketPoint data) {
+        if (null == data || null == data.getSamples() || 0 == data.getSamples()) {
+            throw new IllegalArgumentException("AvailabilityBucketPoint has no samples");
+        }
+
+        try {
+            switch (Function.valueOf(func)) {
+                case notUpCount:
+                    return data.getNotUpCount().doubleValue();
+                case notUpDuration:
+                    return data.getNotUpDuration().doubleValue();
+                case upCount:
+                    return data.getUpCount().doubleValue();
+                case uptimeRatio:
+                    return data.getUptimeRatio().doubleValue();
+                case samples:
+                    return data.getSamples() * 1.0;
+                default:
+                    throw new IllegalArgumentException("Unexpected func [" + func + "]");
+            }
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid value for func [" + func + "] in " + data.toString());
+        }
+    }
+
+    Percentile funcToPercentile(String func, List<Percentile> percentiles) {
+        String quantile = func.substring(1);
+        for (Percentile p : percentiles) {
+            if (p.getOriginalQuantile().equals(quantile)) {
+                return p;
+            }
+        }
+        throw new IllegalArgumentException("Failed to find Percentile for [" + func + "] in " + percentiles);
+    }
+
+    public Map<String, QueryFunc> getQueryVars() {
+        return queryVars;
+    }
+
+    public Expression getExpression() {
+        return expression;
+    }
+
+    private String replaceQueriesWithVariables(String eval) {
+        if (null == queryVars) {
+            queryVars = new HashMap<>();
+        }
+        queryVars.clear();
+
+        // First, collect the queries
+        List<String> queries = new ArrayList<String>();
+        Matcher m = Pattern.compile("q\\(.*?\\)").matcher(eval);
```

Comment:
Minor optimization.
If this one pattern is going to be used often you could make it static as part of the class.